src/lib/sample_sheet.rs

```diff
@@ -209,31 +228,67 @@ impl SampleSheet {
             &opts.undetermined_sample_name,
         )?;
 
-        Ok(SampleSheet { samples, opts: opts.clone() })
+        Ok(SampleSheet { samples, opts })
     }
 
-    // Updates the given command line options with values from the `[Demux]` section.
-    // Starting at the given line index, read in any demultiplexing command line options and
-    // values.  The option names must be the same name as on the command line, with the leading
-    // `--` ommited.  If the value is empty, the command line option is assumed to be a flag.
-    // Command line options with empty values are not supported.  The given opts are updated.
-    fn slurp_demux_opts(
-        records: &[StringRecord],
-        start_line_index: usize,
-        opts: &mut Opts,
-    ) -> Result<usize, SampleSheetError> {
-        let mut line_index = start_line_index;
-        if records.len() == line_index {
-            // no "[Demux]" found
-            return Err(SampleSheetError::NoDemuxHeader);
+    /// Finds the start and end line index (0-based exclusive) of the section with the given key
+    /// (e.g.  "[Header]", "[Demux]", "[Data]"), returning `None` if the key wasn't found.  The key
+    /// must occur in the first column.  The end of the section is identified by the first column
+    ///  starting with "[".
+    ///
+    /// Arguments:
+    /// - `records` - the `StringRecord`s, one per line
+    /// - `section_key` - the value of the first column that identifies the start of the section
+    fn find_section(records: &[StringRecord], section_key: &str) -> Option<(usize, usize)> {
+        let mut start_line_index = 0;
+        while start_line_index < records.len() {
+            let record = &records[start_line_index];
+            start_line_index += 1;
```

Comment:
Won't this cause it to report the line _after_ the one you want?  E.g. if `[Demux]` is on the first line, you'll increment start_line_index before you check for it, so you'll get 1 not 0?

Reply:
Oh, or is the intent to return the lines in the section without the header?  That would make sense, but should probably be in the doc-string as it took me a minute.